lib/chef/resource/registry_key.rb

```diff
@@ -118,11 +118,38 @@ class RegistryKey < Chef::Resource
       Note: Be careful when using the :delete_key action with the recursive attribute. This will delete the registry key, all of its values and all of the names, types, and data associated with them. This cannot be undone by Chef Infra Client.
       DOC
 
-      state_attrs :values
-
       default_action :create
       allowed_actions :create, :create_if_missing, :delete, :delete_key
 
+      VALID_VALUE_HASH_KEYS = %i{name type data}.freeze
+
+      property :key, String, name_property: true
+      property :values, [Hash, Array],
+        default: [],
+        coerce: proc { |v|
+          case v
+          when Hash
+            @unscrubbed_values = [ Mash.new(v).symbolize_keys ]
+          when Array
+            @unscrubbed_values = v.map { |value| Mash.new(value).symbolize_keys }
+          else
+            @unscrubbed_values = []
+          end
```

Comment:
I think you can go back to the `raise` now after this refactoring, I don't see any particularly clean way to do this and fall through to the property validation.

Should really pull the case statement out to the right hand side of the assignment:

```
@unscrubbed_values = case v ...
```

Reply:
or:

```
@unscrubbed_values =
  case v
  end
```

to avoid excessive indentation